### PR TITLE
Trunk-5063 Extract private method in ConceptServiceImpl.saveConcept

### PR DESCRIPTION
**Pull request by meenakshi-dhanani**
I worked on this bug, and made all changes mentioned in the description.
-Extracted method - ensureConceptMapTypeIsSet, ensurePreferredNameForLocale
-Inline a temporary variable
Did not add any tests, only refactored 

I also completed the following: 
- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.
- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] All new and existing **tests passed**.
- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by dkayiwa** [comment]
@meenakshi-dhanani do you mind squashing the commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by meenakshi-dhanani** [comment]
Sorry for that. I have squashed all into one now.

**Comment by meenakshi-dhanani** [comment]
Is it okay or do I need to make more changes for this ticket?

File: api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java [hunk 1]
```diff
@@ -216,27 +234,14 @@ public Concept saveConcept(Concept concept) throws APIException {
...
+
 			if (possiblePreferredName != null) { //there may have been none
 				possiblePreferredName.setLocalePreferred(true);
 			}
 			checkedLocales.add(locale);
 		}
-		
-		//See TRUNK-3337 for why we set changed by and date changed every time we save a concept.
-		concept.setDateChanged(new Date());
-		concept.setChangedBy(Context.getAuthenticatedUser());
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: Was moving this part of the ticket goals?

**Comment by meenakshi-dhanani** [comment]
No, but I think because i extracted two methods those lines moved to the
top.

File: api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java (hunk 1 above)
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: Can you do only what the ticket is all about and no more? Not formatting, no nothing else. It greatly simplifies the review. 😊

Comment (meenakshi-dhanani) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: This is part of the ticket I guess, I have not made any formatting changes. When I extract the lines into another method, these lines move up. 
Extracting the ensurePreferredNameForLocale is part of this ticket. Sorry for the trouble.

Comment (meenakshi-dhanani) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: I have just made the changes mentioned in the ticket. Is it okay?

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: Did you commit and push the change to this pull request?

Comment (meenakshi-dhanani) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: Yes I squashed the commits and rechecked if I have made any formatting changes.

**Pull request merged**